packages/next/client/image.tsx

```diff
@@ -698,54 +744,129 @@ export default function Image({
+    lazy: isLazy,
+    ...{
+      imgAttributes,
+      heightInt,
+      widthInt,
+      qualityInt,
+      layout,
+      className,
+      imgStyle,
+      blurStyle,
+      imgRef,
+      loading,
+      config,
+      unoptimized,
+      loader,
+    },
+    srcString: srcString,
+    ...rest,
+  }
   return (
-    <span style={wrapperStyle}>
-      {hasSizer ? (
-        <span style={sizerStyle}>
-          {sizerSvgUrl ? (
-            <img
-              style={{
-                display: 'block',
-                maxWidth: '100%',
-                width: 'initial',
-                height: 'initial',
-                background: 'none',
-                opacity: 1,
-                border: 0,
-                margin: 0,
-                padding: 0,
-              }}
-              alt=""
-              aria-hidden={true}
-              src={sizerSvgUrl}
-            />
+    <>
+      {layout === 'raw' ? (
+        <ImageElement raw {...imgElementArgs} />
+      ) : (
+        <span style={wrapperStyle}>
+          {hasSizer ? (
+            <span style={sizerStyle}>
+              {sizerSvgUrl ? (
+                <img
+                  style={{
+                    display: 'block',
+                    maxWidth: '100%',
+                    width: 'initial',
+                    height: 'initial',
+                    background: 'none',
+                    opacity: 1,
+                    border: 0,
+                    margin: 0,
+                    padding: 0,
+                  }}
+                  alt=""
+                  aria-hidden={true}
+                  src={sizerSvgUrl}
+                />
+              ) : null}
+            </span>
           ) : null}
+          <ImageElement raw={false} {...imgElementArgs} />
         </span>
+      )}
+      {priority ? (
+        // Note how we omit the `href` attribute, as it would only be relevant
+        // for browsers that do not support `imagesrcset`, and in those cases
+        // it would likely cause the incorrect image to be preloaded.
+        //
+        // https://html.spec.whatwg.org/multipage/semantics.html#attr-link-imagesrcset
+        <Head>
+          <link
+            key={
+              '__nimg-' +
+              imgAttributes.src +
+              imgAttributes.srcSet +
+              imgAttributes.sizes
+            }
+            rel="preload"
+            as="image"
+            href={imgAttributes.srcSet ? undefined : imgAttributes.src}
+            {...linkProps}
+          />
+        </Head>
       ) : null}
+    </>
+  )
+}
+
+const ImageElement = ({
+  raw,
+  imgAttributes,
+  heightInt,
+  widthInt,
+  qualityInt,
+  layout,
+  className,
+  imgStyle,
+  blurStyle,
+  lazy,
+  imgRef,
+  loading,
+  sizes,
+  srcString,
+  config,
+  unoptimized,
+  loader,
+  ...rest
+}: ImageElementProps) => {
+  return (
+    <>
       <img
         {...rest}
         {...imgAttributes}
+        {...(raw ? { height: heightInt, width: widthInt } : {})}
```

Comment:
If `layout=raw` is going to pass through `width` and `height`, what is the purpose of `sizes`? Seems like we could use the `1x`, `2x` srcset instead?

Reply:
You could use the height/width for setting the aspect ratio so the browser knows what space to reserve to reduce CLS.

The browser will try to optimize before any sizer or padding trick. From my experience, the CLS is still there, but almost zero. The SVG sizer and padding tricks are really only needed to reduce your already low 0.05 CLS to 0...

I'm not 100% sure on this one, but if you keep the height/width, I think the browser will use the calculated aspect ratio and the available space to choose the right version to download from the `srcset`.

What I'm 100% sure, removing the height/width will not benefit you in any way. Furthermore, people using the `raw` layout are most likely gonna want to control their image entirely, which means removing a value they specifically passed is most likely not wanted.

Reply:
What it's doing now, which I think is probably the best compromise, is adding height and width unless you use `sizes`. Sizes should only be used for responsive images, so it doesn't really make sense to include a fixed height and width. Also, having those height and width attributes on a responsive image can mess up a user's intended responsive behavior. For instance, if the user sets a responsive width without realizing that there's a fixed height still attached to the image, it will stretch out the image.

Finally, I'm now adding a `aspect-ratio` style to all raw images. As discussed in the past this still misses a couple of supported browsers, but it should go a long way toward preventing CLS in the responsive case.

TLDR: Images using raw layout with a user-provided responsive styling could (possibly but not likely) suffer from CLS on ie11 and old versions of other browsers. The share of users on such browsers will decrease over time.